libraries/src/Updater/Adapter/TufAdapter.php

```diff
@@ -0,0 +1,211 @@
+<?php
+
+/**
+ * Joomla! Content Management System
+ *
+ * @copyright  (C) 2024 Open Source Matters, Inc. <https://www.joomla.org>
+ * @license    GNU General Public License version 2 or later; see LICENSE.txt
+ */
+
+namespace Joomla\CMS\Updater\Adapter;
+
+// phpcs:disable PSR1.Files.SideEffects
+\defined('_JEXEC') or die;
+// phpcs:enable PSR1.Files.SideEffects
+
+use Joomla\CMS\Application\ApplicationHelper;
+use Joomla\CMS\Factory;
+use Joomla\CMS\Table\Table;
+use Joomla\CMS\Table\Tuf as MetadataTable;
+use Joomla\CMS\TUF\TufFetcher;
+use Joomla\CMS\Updater\ConstraintChecker;
+use Joomla\CMS\Updater\UpdateAdapter;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+use Tuf\Exception\MetadataException;
+
+/**
+ * TUF Update Adapter Class
+ *
+ * @since   __DEPLOY_VERSION__
+ */
+class TufAdapter extends UpdateAdapter
+{
+    /**
+     * Finds an update.
+     *
+     * @param array $options Update options.
+     *
+     * @return  array|boolean  Array containing the array of update sites and array of updates. False on failure
+     *
+     * @since   __DEPLOY_VERSION__
+     */
+    public function findUpdate($options)
+    {
+        $updates = [];
+        $targets = $this->getUpdateTargets($options);
+
+        if ($targets) {
+            foreach ($targets as $target) {
+                $updateTable = Table::getInstance('update');
+                $updateTable->set('update_site_id', $options['update_site_id']);
+
+                $updateTable->bind($target);
+
+                $updates[] = $updateTable;
+            }
+        }
+
+        return ['update_sites' => [], 'updates' => $updates];
+    }
+
+    /**
+     * Finds targets.
+     *
+     * @param array $options Update options.
+     *
+     * @return  array|boolean  Array containing the array of update sites and array of updates. False on failure
+     *
+     * @since   __DEPLOY_VERSION__
+     */
+    public function getUpdateTargets($options)
+    {
+        $versions = [];
+
+        /** @var MetadataTable $metadataTable */
+        $metadataTable = new MetadataTable(Factory::getDbo());
```

Comment:
```suggestion
        $metadataTable = new MetadataTable($this->db);
```
The database object can be accessed via `db` property, so we should not use deprecated Factory::getDbo(). If you make this change, the `use Joomla\CMS\Factory;`  statement at the top could be removed, too.

Reply:
Done